sklearn/metrics/_plot/tests/test_cap_curve_display.py

```diff
@@ -0,0 +1,537 @@
+    [
+        ("from_estimator", "LogisticRegression"),
+        ("from_predictions", "Classifier"),
+    ],
+)
+def test_cap_curve_valid_position(
+    pyplot,
+    data_binary,
+    response_method,
+    constructor_name,
+    default_name,
+    with_sample_weight,
+    with_strings,
+    pos_label,
+):
+    X, y = data_binary
+
+    if with_strings:
+        y = np.array(["a", "b"])[y]
+
+    if with_sample_weight:
+        rng = np.random.RandomState(42)
+        sample_weight = rng.randint(1, 4, size=(X.shape[0]))
+    else:
+        sample_weight = None
+
+    lr = LogisticRegression()
+    lr.fit(X, y)
+
+    if response_method == "auto":
+        response_method = "predict_proba"
+
+    y_pred = getattr(lr, response_method)(X)
+    if y_pred.ndim != 1:
+        if pos_label is not None:
+            class_index = np.where(lr.classes_ == pos_label)[0][0]
+        else:
+            class_index = -1
+        y_pred = y_pred[:, class_index]
+    else:
+        if pos_label is not None:
+            if pos_label != lr.classes_[1]:
+                y_pred = -y_pred
+
+    if constructor_name == "from_estimator":
+        display = CAPCurveDisplay.from_estimator(
+            lr,
+            X,
+            y,
+            pos_label=pos_label,
+            sample_weight=sample_weight,
+            alpha=0.8,
+        )
+    else:
+        display = CAPCurveDisplay.from_predictions(
+            y,
+            y_pred,
+            pos_label=pos_label,
+            sample_weight=sample_weight,
+            alpha=0.8,
+        )
+
+    assert display.estimator_name == default_name, (
+        f"{display.estimator_name} != {default_name}"
+    )
+
+    cumulative_total = display.cumulative_total
+    y_true_cumulative = display.y_true_cumulative
+    assert np.all(y_true_cumulative >= cumulative_total - 1e-8), (
+        f"CAP curve dips below the chance line.\n"
+        f"x: {cumulative_total}\n"
+        f"y: {y_true_cumulative}"
+    )
+
+    perfect_y = np.interp(
+        display.cumulative_total, display._perfect_x, display._perfect_y
+    )
+    assert np.all(y_true_cumulative <= perfect_y + 1e-8), (
+        "CAP curve rises above the perfect prediction line.\n"
+        f"x: {perfect_y}\n"
+        f"y: {y_true_cumulative}"
+    )
+
+
+def test_lorenz_curve_valid_position(pyplot):
+    estimator = PoissonRegressor()
+
+    X, y = make_regression(n_samples=100, n_features=2, noise=10.0, random_state=42)
+    y = y - y.min() + 1e-3
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.4, random_state=42
+    )
+    estimator.fit(X_train, y_train)
+    y_pred = estimator.predict(X_test)
+
+    display = CAPCurveDisplay.from_predictions(y_test, y_pred)
+
+    cumulative_total = display.cumulative_total
+    y_true_cumulative = display.y_true_cumulative
+    assert np.all(cumulative_total >= y_true_cumulative - 1e-8), (
+        f"Lorenz curve dips above the chance line.\n"
```

Comment:
"dips below" or "rises above" depending on the value of the prediction ordering parameter. "dips above" seems contradictory.